src/client/dfs/dfs.c

```diff
@@ -1622,25 +1627,30 @@ open_sb(daos_handle_t coh, bool create, bool punch, int omode, daos_obj_id_t sup
 
 	/** check if SB info exists */
 	if (iods[MAGIC_IDX].iod_size == 0) {
-		D_ERROR("SB does not exist.\n");
-		D_GOTO(err, rc = ENOENT);
+		rc = ENOENT;
```

Comment:
I'm not sure if ENOENT is the correct error to return here.  If you create a posix container without a SB then you get here and ENOTENT could be mistaken for the container doesn't exist - whereas in this case the container does but it's not usable so EIO might be more suitable.

Reply:
i can make an argument with or against each of those error codes